src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,651 @@
+from ..convergence import forward_backward_convergence
+from .. import concat
+from .. import __version__
+
+
+class ABFE():
+    '''Alchemical Analysis style automatic workflow.
+
+    Parameters
+    ----------
+    units : str
+        The unit used for printing and plotting results. {'kcal/mol', 'kJ/mol',
+        'kT'}. Default: 'kT'.
+    software : str
+        The software used for generating input. {'Gromacs', }
+    dir : str
+        Directory in which data files are stored. Default: os.path.curdir.
+    prefix : str
+        Prefix for datafile sets. Default: 'dhdl'.
+    suffix : str
+        Suffix for datafile sets. Default: 'xvg'.
+    T : float
+        Temperature in K. Default: 298.
+    skiptime : float
+        Discard data prior to this specified time as 'equilibration' data. Units
+        picoseconds. Default: 0.
+    uncorr : str
+        The observable to be used for the autocorrelation analysis; 'dhdl'
+        (obtained as a sum over those energy components that are changing).
+        Default: `dhdl`
+    threshold : int
+        Proceed with correlated samples if the number of uncorrelated samples is
+        found to be less than this number. If 0 is given, the time series
+        analysis will not be performed at all. Default: 50.
+    methods : str
+        A list of the methods to esitimate the free energy with. Default: None.
+    out : str
+        Directory in which the output files produced by this script will be
+        stored. Default: os.path.curdir.
+    overlap : str
+        The filename for the plot of overlap matrix. Default: None. (not
+        plotting).
+    breakdown : bool
+        Plot the free energy differences evaluated for each pair of adjacent
+        states for all methods, including the dH/dlambda curve for TI. Default:
+        None. (not plotting).
+    forwrev : int
+        Plot the free energy change as a function of time in both directions,
+        with the specified number of points in the time plot. The number of time
+        points (an integer) must be provided. Default: None. (not doing
+        convergence analysis).
+    log : str
+        The filename of the log file. The workflow logs under
+        alchemlyb.workflows.ABFE. Default:
+        'result.log'
+
+    Attributes
+    ----------
+    logger : Logger
+        The logging object.
+    file_list : list
+        The list of filenames sorted by the lambda state.
+    u_nk_list : list
+        The list of u_nk read from the files.
+    dHdl_list : list
+        The list of dHdl read from the files.
+    '''
+    def __init__(self, units='kT', software='Gromacs', dir=os.path.curdir,
+                 prefix='dhdl', suffix='xvg', T=298, skiptime=0, uncorr=None,
+                 threshold=50, methods=None, out=os.path.curdir,
+                 overlap=None, breakdown=None, forwrev=None,
+                 log='result.log'):
+
+        logging.basicConfig(filename=log, level=logging.INFO)
```

Comment:
I don't think that you should override the logging config that may be set by calling code. Remove the config.

Users need to configure their logger. You can add a `alchemlyb.start_logging()` function (see `MDAnalysis.start_logging()` and add docs. alchemlyb is a library first, it does not have to do everything for the user. flamel can set its logger.